src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -609,6 +609,7 @@ internal static bool WriteConsideredUse(TypeSymbol type, BoundExpression value)
                         return WriteConsideredUse(null, boundConversion.Operand);
                     }
                 case BoundKind.DefaultOperator:
+                case BoundKind.ConvertedTupleLiteral:
```

Comment:
The intent of the warning suppression is that the warning should be suppressed if the user needs the assignment of the local to get some effect, for example a user-defined conversion, or to hold on to memory, etc. These conditions should be recursively applied to the contents of the converted tuple literal to get corresponding semantics applied.